### PR TITLE
Fix pagination for DurationBreakdownInsight

### PR DESCRIPTION
**Pull request by MykolaYaremchuk**
Fixed pagination issue for DurationBreakdown insight

### CODE REVIEW

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/editor/PaginationUtil.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import java.util.concurrent.ConcurrentHashMap
+
+private val insightsPaginationInfo: ConcurrentHashMap<String, ConcurrentHashMap<String, Int>> = ConcurrentHashMap()
+
+fun addInsightPaginationInfo(focusedDocumentName: String, uniqueInsightId: String, activePage: Int) {
+    if (activePage > 1) {
+        val insightsPaginationInfosMapForActualFile = insightsPaginationInfo.get(focusedDocumentName)
+        if (insightsPaginationInfosMapForActualFile != null) {
+            insightsPaginationInfosMapForActualFile[uniqueInsightId] = activePage
+            insightsPaginationInfo[focusedDocumentName] = insightsPaginationInfosMapForActualFile
```
Comment (asafchen-dig) [inline ide-common/src/main/kotlin/org/digma/intellij/plugin/editor/PaginationUtil.kt:12]: Is this line really needed?
`insightsPaginationInfo[focusedDocumentName] = insightsPaginationInfosMapForActualFile`

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/editor/PaginationUtil.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            filesAndInsightsPaginationInfosMap[uniqueInsightId] = activePage
+            insightsPaginationInfo[focusedDocumentName] = filesAndInsightsPaginationInfosMap
+        }
+    } else if (insightsPaginationInfo.contains(uniqueInsightId)) {
+        // remove pagination info for insights which had some page selected before
+        insightsPaginationInfo.remove(uniqueInsightId)
+    }
+}
+
+fun getInsightPaginationInfo(uniqueInsightId: String): Int? {
```
Comment (asafchen-dig) [inline ide-common/src/main/kotlin/org/digma/intellij/plugin/editor/PaginationUtil.kt:24]: Shouldn't it take into account the `focusedDocumentName` ?

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/editor/FocusedDocumentHelper.kt [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package org.digma.intellij.plugin.editor
+
+private var openedFilesList: MutableList<String> = ArrayList()
```
Comment (asafchen-dig) [inline ide-common/src/main/kotlin/org/digma/intellij/plugin/editor/FocusedDocumentHelper.kt:3]: Who uses/query the `openedFilesList` ? I see only `add` and `remove`

Comment (MykolaYaremchuk) [reply ide-common/src/main/kotlin/org/digma/intellij/plugin/editor/FocusedDocumentHelper.kt:3]: great catch! thanks! I've just remove it. it's no longer needed

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/editor/PaginationUtil.kt (hunk 1 above)
Comment (MykolaYaremchuk) [reply ide-common/src/main/kotlin/org/digma/intellij/plugin/editor/PaginationUtil.kt:12]: it's not needed as well :)

File: ide-common/src/main/kotlin/org/digma/intellij/plugin/editor/PaginationUtil.kt (hunk 2 above)
Comment (MykolaYaremchuk) [reply ide-common/src/main/kotlin/org/digma/intellij/plugin/editor/PaginationUtil.kt:23]: thank you for useful advice!

**Review by asafchen-dig — APPROVED** [review]

**Pull request merged**